aten/src/ATen/native/MetaTensor.cpp

```diff
@@ -29,15 +29,22 @@ Tensor empty_meta_symint(
       size, dtype_opt, layout_opt, device_opt, pin_memory_opt, memory_format_opt);
 }
 
-Tensor empty_strided_meta(
-  IntArrayRef size,
-  IntArrayRef stride,
+Tensor empty_strided_meta_symint(
+  SymIntArrayRef size,
+  SymIntArrayRef stride,
   c10::optional<ScalarType> dtype_opt,
   c10::optional<Layout> layout_opt,
   c10::optional<Device> device_opt,
   c10::optional<bool> pin_memory_opt
 ) {
-  return at::detail::empty_strided_meta(
+  auto opt_size = asIntArrayRefSlowOpt(size);
+  auto opt_stride = asIntArrayRefSlowOpt(stride);
+  if (opt_size.has_value() && opt_stride.has_value()) {
+    TORCH_INTERNAL_ASSERT(opt_size->size() == opt_stride->size());
+    return at::detail::empty_strided_meta(
+        *opt_size, *opt_stride, dtype_opt, layout_opt, device_opt, pin_memory_opt);
+  }
```

Comment:
It would be nice to not have to do this "optimization" anymore

Reply:
This was mostly cargo-culted from the pattern in `at::native::empty_meta_symint`. Do you mean it would be nicer if we didn't need a separate kernel for meta?

Reply:
the optimization here is routing to `empty_strided_meta` if it turns out there are no symints in the args. But this shouldn't be necessary if the two implementations are equivalent.

Reply:
So can I just try deleting it?